Sources/String+Sweetness.swift

```diff
@@ -1,5 +1,34 @@
 import Foundation
 
+public protocol _StringIndex {}
+extension String.Index: _StringIndex {}
+
+public extension Range where Bound: _StringIndex {
+    /// Translates a Range<String.Index> into a correct NSRange for use with NSStrings.
+    /// Should help avoind all the `as String` or `as NSString` casting.
+    ///
+    /// - Parameter string: string where ranges will be applied. Necessary to ensure that multi-byte code-points are treated properly.
+    /// - Returns: the NSRange representation of a Range<String.Index> for the given string.
+    public func nsRange(on string: String) -> NSRange {
+        let lowerBound = self.lowerBound as! String.Index
+        let upperBound = self.upperBound as! String.Index
```

Comment:
why force unwrap instead of `guard let` and returning some kind of garbage value

Reply:
I went from the principle that this should never, ever fail. 🤔 I didn't want to make this throwable or do some other sort of fix that could go unnoticed by a user. Let's say it does break somehow, and we return a garbage `NSRange` value. Now we can have it breaking code somewhere else, far away from this, when the user tries to split a string or something. How to trace it back to us easily? If the method itself crashes, I find that more useful. If we can crash with some explanation that would be ideal.

Reply:
You could do `guard let` and throw a `fatalError` with a custom error message.

Reply:
Do they actually show up outside of `debug`? 👍

Reply:
`fatalError` will always crash your app, even in production (ask me how I know!). `assertionFailure` is what gets no-op'd when you're in release mode.

Reply:
I know it crashes, but does it crash with the message? I think I recall the message getting lost, never bubbling up to the caller.

Reply:
In my current experience the message is recorded to Crashlytics.

Reply:
Great! 🚀